Source/NameServer/AddressBook.cpp

```diff
@@ -0,0 +1,1361 @@
+//==============================================================================
+#include "AddressBook.hpp"
+//#include "Debug.h"
+
+#include <iostream>
+
+//==============================================================================
+namespace AddressBookNS
+{
+#ifdef EXCEPTS
+#define ERETURN(X, Y) throw(X)
+#else
+#define ERETURN(X, Y) return Y
+#endif
+
+// static constant definitions
```

Comment:
Use an enum.

Reply:
This was mine when I was considering having throwable exceptions.
I meant to talk to you about your preference - The exceptions really did not work well with Borland (it insisted on stepping down the exception tree) hence the return code option.

Reply:
I think this has to do with https://github.com/heliosfa's commonets above

Reply:
Mark's comment was in regard to the following list of consts.

Reply:
@heliosfa has it right here.